code/modules/mining/mint.dm

```diff
@@ -1,198 +1,243 @@
+		var/obj/item/stack/sheet/sh = new ma.sheettype(out_T)
+		sh.amount = materials.getAmount(href_list["eject"])
+		materials.removeAmount(href_list["eject"], sh.amount)
+		if (chosen == href_list["eject"])
+			chosen = null
+
+	if("changedir" in href_list)
+		//Change_Dir()
+		var/changingdir = text2num(href_list["changedir"]) //See ore processing_unit for original comments
+		changingdir = Clamp(changingdir, 1, 2)
+
+		var/newdir = input("Select the new direction", name, "North") as null|anything in list("North", "South", "East", "West")
+		if(!newdir)
+			return 1
+		newdir = text2dir(newdir)
+
+		var/list/dirlist = list(in_dir, out_dir)
+		var/olddir = dirlist[changingdir]
+		dirlist[changingdir] = -1
+
+		var/conflictingdir = dirlist.Find(newdir)
+		if(conflictingdir)
+			dirlist[conflictingdir] = olddir
+
+		dirlist[changingdir] = newdir
+
+		in_dir = dirlist[1]
+		out_dir = dirlist[2]
+
 	if(href_list["choose"])
 		chosen = href_list["choose"]
+
 	if(href_list["chooseAmt"])
 		coinsToProduce = Clamp(coinsToProduce + text2num(href_list["chooseAmt"]), 0, 1000)
+
 	if(href_list["makeCoins"])
+		if(chosen == null)
+			return
 		var/temp_coins = coinsToProduce
-		if (src.output)
+		if (src.out_dir)
 			processing = 1
 			icon_state = "coinpress1"
-			//var/obj/item/weapon/storage/bag/money/M
 			var/datum/material/po=materials.getMaterial(chosen)
 			if(!po)
 				chosen=null
 				processing=0
 				return
 			while(materials.storage[chosen] > 0 && coinsToProduce > 0)
-			/*	if (locate(/obj/item/weapon/storage/bag/money,output.loc))
-					M = locate(/obj/item/weapon/storage/bag/money,output.loc)
-					if(M.can_be_inserted(po.cointype, 1))
-						new po.cointype(M)
-					else
-						new po.cointype(output.loc)
-				else
-				//Can't seem to be able to get the can_be_inserted check to work, would always drop the coin at the output loc
-				*/
-					//M = new/obj/item/weapon/storage/bag/money(output.loc)
-				new po.cointype(output.loc)
-				materials.removeAmount(chosen, 1)
-				coinsToProduce--
-				newCoins++
-				src.updateUsrDialog()
-				sleep(5)
+				var/obj/item/weapon/storage/bag/money/tempbag = locate(/obj/item/weapon/storage/bag/money,out_T)
+				materials.removeAmount(chosen, 1) //We'll get that money up front don't you worry.
+				for(var/i=0,i<coins_per_sheet,i++)
+					var/obj/item/weapon/coin/co = new po.cointype(out_T)
+					if(tempbag)
+						if(tempbag.can_be_inserted(co, 1))
+							tempbag.handle_item_insertion(co, 1)
+					coinsToProduce--
+					newCoins++
+					src.updateUsrDialog()
+					sleep(2)
+				sleep(2)
 			icon_state = "coinpress0"
 			processing = 0
 			coinsToProduce = temp_coins
 	src.updateUsrDialog()
-	return
\ No newline at end of file
+	return
+
+/obj/machinery/mineral/mint/Destroy()
+	qdel(mover)
+	mover = null
+	..()
+
+/obj/machinery/mineral/mint/crowbarDestroy(mob/user)
+	if(..() == 1)
+		if (materials)
+			for(var/matID in materials.storage)
+				DropSheet(matID)
+	return 1
```

Comment:
this is wrong
I suggest you do something like
```
. = ..()
if(.)
    if(materials)
        for(var/matID in materials.storage)
            DropSheet(matID)
```

as it is, you're always returning 1 and that means it will be deleted even if you interrupt the action, which is not what you want